modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java

```diff
@@ -85,15 +85,19 @@ public Hits search(SearchContext searchContext, Query query) {
 
 			SearchRequest searchRequest = _getSearchRequest(searchContext);
 
-			Integer from = searchRequest.getFrom();
 			Integer size = searchRequest.getSize();
 
-			if ((from == null) && (size != null) && (size > end)) {
+			if ((size != null) && (size > end)) {
 				end = size;
+			}
+
+			Integer from = searchRequest.getFrom();
+
+			if (from == null) {
 				start = 0;
 			}
-			else if ((from != null) && (size != null) && (size > end)) {
-				end = from + size;
+			else {
+				end = from + end;
 				start = from;
 			}
```

Comment:
Shouldn't it be?

	if ((size != null) && (size > end)) {
				if ((from == null)) {
					end = size;
					start = 0;
				}
				else {
					end = from + size;
					start = from;
				}
			}

Reply:
@joshuacords

Reply:
@gustavolimav, I think you're right. But why exactly we don't ever run `end = from + end` makes me uncomfortable. In general the `end` variable is overloaded here. I think this is the simplest fix but I'm not sure it's the best.